packages/cheqd/src/dids/CheqdDidResolver.ts

```diff
@@ -0,0 +1,189 @@
+import type { ParsedCheqdDid } from '../anoncreds/utils/identifiers'
+import type { AgentContext, DidDocument, DidResolutionResult, DidResolver, ParsedDid } from '@aries-framework/core'
+import type { Metadata } from '@cheqd/ts-proto/cheqd/resource/v2'
+
+import { AriesFrameworkError } from '@aries-framework/core'
+
+import {
+  cheqdDidMetadataRegex,
+  cheqdDidRegex,
+  cheqdDidVersionRegex,
+  cheqdDidVersionsRegex,
+  cheqdResourceMetadataRegex,
+  cheqdResourceRegex,
+  parseCheqdDid,
+} from '../anoncreds/utils/identifiers'
+import { CheqdLedgerService } from '../ledger'
+
+import { filterResourcesByNameAndType, getClosestResourceVersion, renderResourceData } from './didCheqdUtil'
+
+export class CheqdDidResolver implements DidResolver {
+  public readonly supportedMethods = ['cheqd']
+
+  public async resolve(agentContext: AgentContext, did: string, parsed: ParsedDid): Promise<DidResolutionResult> {
```

Comment:
We are gonna to support a lot of queries for resolving DID and resources BTW. Should we maybe create a base for future adds? I mean in case of supporting more then 10 queries it will be really big switch, no?

Reply:
For the use case of anoncreds, only these four queries are relevant and used, Mainly to query a resource using name, type and the resourceTime. So not sure if we need to complicate it here